### PR TITLE
Login form

### PR DESCRIPTION
**Pull request by prestys**

### CODE REVIEW

File: api/controllers/tokens.js
Comment (RichASMarshall) [inline api/controllers/tokens.js:1]: We will add tests for this as a separate ticket.

File: api/controllers/users.js
Comment (RichASMarshall) [inline api/controllers/users.js:1]: We will add more tests for this in another ticket.

File: api/models/token_generator.js
Comment (RichASMarshall) [inline api/models/token_generator.js:1]: We will take account of these tests in a different ticket.

File: api/routes/[route]/index.js
Comment (RichASMarshall) [inline api/routes/[route]/index.js:1]: Does this console/log need to remain here?

File: frontend/next.config.js
Comment (RichASMarshall) [inline frontend/next.config.js:1]: This is a great solution!! :)

File: frontend/src/Login/Login.tsx
Comment (RichASMarshall) [inline frontend/src/Login/Login.tsx:1]: Were you getting TypeScript errors for your event listeners here? If so, let us know and we will sort these with a new ticket.

File: frontend/src/Navbar/Navbar.tsx
Comment (RichASMarshall) [inline frontend/src/Navbar/Navbar.tsx:1]: This looks awesome!

File: frontend/src/app/page.tsx
Comment (RichASMarshall) [inline frontend/src/app/page.tsx:1]: Is it the case that the navbar wants to live in the layout file?

**Review by RichASMarshall — CHANGES_REQUESTED** [review]
Overall awesome work. Just a few things to look at in the comments.

**Review by prestys — COMMENTED** [review]
Done changes

**Review by jpthacker — APPROVED** [review]
Nicely done!

**Pull request merged**